pkg/function/batch.go

```diff
@@ -25,6 +25,10 @@ const (
 )
 
 func (s *EdgeRuntimeAPI) UpsertFunctions(ctx context.Context, functionConfig config.FunctionConfig, filter ...func(string) bool) error {
+	return s.upsertFunctions(ctx, functionConfig, false, filter...)
+}
+
+func (s *EdgeRuntimeAPI) upsertFunctions(ctx context.Context, functionConfig config.FunctionConfig, dryRun bool, filter ...func(string) bool) error {
```

Comment:
Similarly for UpsertFunctions, I'd prefer we continue using filter variadic arg instead of introducing a new dryRun arg.

Reply:
All the dry-run functionality should use the variadic filter param but let me know if my approach isn't what you had in mind